mirrord/agent/src/sniffer.rs

```diff
@@ -199,78 +210,56 @@ pub struct SnifferCommand {
 pub struct TcpSnifferApi {
     client_id: ClientId,
     sender: Sender<SnifferCommand>,
-    pub receiver: Receiver<DaemonTcp>,
+    receiver: Receiver<DaemonTcp>,
+    /// View on the sniffer task's status.
+    task_status: TaskStatus,
 }
 
 impl TcpSnifferApi {
     pub async fn new(
         client_id: ClientId,
         sniffer_sender: Sender<SnifferCommand>,
-        receiver: Receiver<DaemonTcp>,
-        tcp_sender: Sender<DaemonTcp>,
+        task_status: TaskStatus,
+        channel_size: usize,
     ) -> Result<TcpSnifferApi, AgentError> {
+        let (sender, receiver) = mpsc::channel(channel_size);
+
         sniffer_sender
             .send(SnifferCommand {
                 client_id,
-                command: SnifferCommands::NewAgent(tcp_sender),
+                command: SnifferCommands::NewAgent(sender),
             })
             .await?;
+
         Ok(Self {
             client_id,
             sender: sniffer_sender,
             receiver,
+            task_status,
         })
     }
 
-    pub async fn subscribe(&mut self, port: Port) -> Result<(), AgentError> {
-        self.sender
-            .send(SnifferCommand {
-                client_id: self.client_id,
-                command: SnifferCommands::Subscribe(port),
-            })
-            .await
-            .map_err(From::from)
-    }
-
-    pub async fn connection_unsubscribe(
-        &mut self,
-        connection_id: ConnectionId,
-    ) -> Result<(), AgentError> {
+    async fn send_command(&self, command: SnifferCommands) -> Result<(), AgentError> {
```

Comment:
Hey Michal, could you add some rust docs for some of the functions (and things in general) you're (already) touching?

This is part of our `#[deny(missing_docs)]` (or whatever the warning is called) initiative.

No need to add docs for everything though, if you're not sure about something just move on and don't waste time (imo).

Reply:
I added some docs, mostly in the sniffer code ^^